Fix compatibility with DBAL develop

Majkl578 opened the pull request:
* ResultStatement signature BC break
* PDO::FETCH_* -> FetchMode::*
* PDO::PARAM_* -> ParameterType::*
* AbstractPlatform::DATE_INTERVAL_UNIT_* -> DateIntervalUnit::*
* AbstractPlatform::TRIM_* -> TrimMode::*

Changes from doctrine/dbal#2958 & doctrine/dbal#2998.

/cc @morozov

Majkl578 committed:
Fix compatibility with DBAL develop

* ResultStatement signature BC break
* PDO::FETCH_* -> FetchMode::*
* PDO::PARAM_* -> ParameterType::*
* AbstractPlatform::DATE_INTERVAL_UNIT_* -> DateIntervalUnit::*
* AbstractPlatform::TRIM_* -> TrimMode::*

morozov commented:
@Majkl578 I see there are some leftovers of PDO in the codebase and documentation:
```
↪ git grep PDO
docs/en/reference/architecture.rst:top of PDO but is not strongly bound to PDO. The purpose of this
docs/en/reference/dql-doctrine-query-language.rst:            return $this->stmt->fetchAll(PDO::FETCH_ASSOC);
docs/en/reference/query-builder.rst:the third argument to ``setParameter()`` explicitly. It accepts either a PDO
docs/en/reference/transactions-and-concurrency.rst:abstraction layers like PDO by default operate in auto-commit mode,
tests/Doctrine/Performance/EntityManagerFactory.php:use Doctrine\DBAL\Driver\PDOSqlite\Driver;
tests/Doctrine/Tests/ORM/Functional/Ticket/DDC1225Test.php:        } catch (\PDOException $e) {
tests/Doctrine/Tests/ORM/Functional/Ticket/DDC1250Test.php:        } catch (\PDOException $e) {
```

1. The documentation shouldn't suggest using PDO symbols.
2. In the tests, if needed at all, the code should be catching a DBAL-level exception, not a PDO one.
3. The `EntityManagerFactory` class relies on the PDO SQLite driver being available which is not always the case.

Majkl578 commented:
Totally missed the docs references. Updated two of them, the other seem legit.
PDOSqlite is used in performance tests to create specific driver, should be fine.
Removed those catches in tests entirely, seems to be a dead code (createSchema catches \Throwable and re-throws ToolsException).
